information/specification.md

```diff
@@ -0,0 +1,705 @@
+  * UTxO<sub>_remaining_</sub> ∩ UTxO<sub>_selected_</sub> = ∅
+  * UTxO<sub>_remaining_</sub> ⋃ UTxO<sub>_selected_</sub> =
+    UTxO<sub>_initial_</sub>
+
+Where:
+
+  * **UTxO<sub>_initial_</sub>**
+
+    is the [initial UTxO set](#initial-utxo-set).
+
+  * **UTxO<sub>_remaining_</sub>**
+
+    is the [remaining UTxO set](#remaining-utxo-set).
+
+  * **UTxO<sub>_selected_</sub>**
+
+    is the value of the _inputs_ field of the [coin selection](#coin-selection)
+    result.
+
+### Conservation of Outputs
+
+This property states that the [requested output list](#requested-output-list)
+is _conserved_ in the [coin selection](#coin-selection) result.
+
+In particular, the _outputs_ field of the [coin selection](#coin-selection)
+result should be _equal to_ the [requested output list](#requested-output-list).
+
+### Cardinality
+
+All algorithms require that:
+
+ 1. Each output from the [requested output list](#requested-output-list) is
+    paid for by _one or more_ entries from the
+    [initial UTxO set](#initial-utxo-set).
+
+ 2. Each entry from the [initial UTxO set](#initial-utxo-set) is used to pay
+    for _at most one_ output from the [requested output
+    list](#requested-output-list).
+
+## Failure Modes
+
+There are a number of ways in which a coin selection algorithm can fail:
+
+  * #### UTxO Balance Insufficient
+
+    This failure occurs when the total value of the entries within the [initial
+    UTxO set](#initial-utxo-set) (the amount of money _available_) is _less
+    than_ the the total value of all entries in the [initial output
+    list](#requested-output-list) (the amount of money _required_).
+
+  * #### UTxO Not Fragmented Enough
+
+    This failure occurs when the _number_ of entries in the [initial UTxO
+    set](#initial-utxo-set) is _smaller than_ the number of entries in the
+    [requested output list](#requested-output-list).
+
+    All algorithms require that there is _at least one_ UTxO entry available
+    _for each_ output.
+
+  * #### UTxO Fully Depleted
+
+    This failure occurs if the algorithm depletes all entries from the [initial
+    UTxO set](#initial-utxo-set) _before_ it is able to pay for all outputs in
+    the [requested output list](#requested-output-list).
+
+  * #### Maximum Input Count Exceeded
+
+    This failure occurs if the _number_ of UTxO entries needed to pay for the
+    outputs in the [requested output list](#requested-output-list) exceeds the
+    upper limit specified by the [maximum input count](#maximum-input-count)
+    parameter.
+
+# Implementations
+
+This section will describe the coin selection algorithms used by Cardano Wallet.
+
+These algorithms implement a [common interface](#interface), as described
+above.
+
+There are two main algorithms used by Cardano Wallet:
+
+  * [Largest-First](#largest-first)
+  * [Random-Improve](#random-improve)
+
+In general, Cardano Wallet gives _priority_ to the
+[Random-Improve](#random-improve) algorithm.
```

Comment:
Perhaps add a short paragraph here explaining *why* Cardano Wallet gives priority to Random-Improve. This paragraph could link to the "Self Organization in Coin Selection" blog article (listed at the end as an external resource).

Reply:
@jonathanknowles wrote:
> 

Perhaps add a short paragraph here explaining why Cardano Wallet gives priority to Random-Improve. This paragraph could link to the "Self Organization in Coin Selection" blog article (listed at the end as an external resource).

Fixed in ebfd135cbbfe2bf7aff749d326f30deca462397f.